spec.html

```diff
@@ -42917,14 +43339,71 @@ <h1>SharedArrayBuffer.prototype.constructor</h1>
         <p>The initial value of `SharedArrayBuffer.prototype.constructor` is %SharedArrayBuffer%.</p>
       </emu-clause>
 
+      <emu-clause id="sec-sharedarraybuffer.prototype.grow">
+        <h1>SharedArrayBuffer.prototype.grow ( _newLength_ )</h1>
+        <p>This method performs the following steps when called:</p>
+        <emu-alg>
+          1. Let _O_ be the *this* value.
+          1. Perform ? RequireInternalSlot(_O_, [[ArrayBufferMaxByteLength]]).
+          1. If IsSharedArrayBuffer(_O_) is *false*, throw a *TypeError* exception.
+          1. Let _newByteLength_ be ? ToIndex(_newLength_).
+          1. Let _hostHandled_ be ? HostGrowSharedArrayBuffer(_O_, _newByteLength_).
+          1. If _hostHandled_ is ~handled~, return *undefined*.
+          1. Let _isLittleEndian_ be the value of the [[LittleEndian]] field of the surrounding agent's Agent Record.
+          1. Let _byteLengthBlock_ be _O_.[[ArrayBufferByteLengthData]].
+          1. Let _currentByteLengthRawBytes_ be GetRawBytesFromSharedBlock(_byteLengthBlock_, 0, ~BigUint64~, *true*, ~SeqCst~).
+          1. Let _newByteLengthRawBytes_ be NumericToRawBytes(~BigUint64~, ℤ(_newByteLength_), _isLittleEndian_).
+          1. Repeat,
+            1. Let _currentByteLength_ be ℝ(RawBytesToNumeric(~BigUint64~, _currentByteLengthRawBytes_, _isLittleEndian_)).
+            1. If _newByteLength_ = _currentByteLength_, return *undefined*.
+            1. If _newByteLength_ &lt; _currentByteLength_ or _newByteLength_ > _O_.[[ArrayBufferMaxByteLength]], throw a *RangeError* exception.
+            1. Let _byteLengthDelta_ be _newByteLength_ - _currentByteLength_.
+            1. If it is impossible to create a new Shared Data Block value consisting of _byteLengthDelta_ bytes, throw a *RangeError* exception.
+            1. NOTE: No new Shared Data Block is constructed and used here. The observable behaviour of growable SharedArrayBuffers is specified by allocating a max-sized Shared Data Block at construction time, and this step is captures the requirement that implementations that run out of memory must throw a *RangeError*.
+            1. Let _readByteLengthRawBytes_ be AtomicCompareExchangeInSharedBlock(_byteLengthBlock_, 0, 8, _currentByteLengthRawBytes_, _newByteLengthRawBytes_).
```

Comment:
I don't love that we inline the Element Size of BigUint64 as 8 in a couple places. Should we refer to the table each time or is it okay to duplicate that?

Reply:
I think it's fine to duplicate because I can't see a possible future where BigUint64 would have a different element size.